### ISSUE TITLE
Short bytes sequence

### ISSUE THREAD

**Issue by lmar76** [avalentino/bpack#2]
Consider the following code:

```python
import sys

import bitstruct
import bpack
import bpack.bs


@bpack.bs.codec
@bpack.descriptor(baseunits='bits', byteorder='>')
class BinaryRecord:
    field_1: int = bpack.field(size=8, signed=False, default=0)
    field_2: int = bpack.field(size=8, signed=False, default=0)


print(f"Python version: {sys.version}")
print(f"bpack version: {bpack.__version__}")
print(f"bitstruct version: {bitstruct.__version__}")

print(f"\nRecord size: {bpack.calcsize(BinaryRecord)}")

for data in (b'\x02\x1a', b'\x02\x1c', b'\x02'):
    print(f"\nData has length: {len(data)}")
    record = BinaryRecord.frombytes(data)
    print(bpack.asdict(record))
```

Executing it on **macOS**, it raises a `ValueError`:

```
Python version: 3.10.5 (v3.10.5:f377153967, Jun  6 2022, 12:36:10) [Clang 13.0.0 (clang-1300.0.29.30)]
bpack version: 0.8.2
bitstruct version: 8.15.1

Record size: 16

Data has length: 2
{'field_1': 2, 'field_2': 26}

Data has length: 2
{'field_1': 2, 'field_2': 28}

Data has length: 1
Traceback (most recent call last):
  File "/Users/luca mariani/Documents/github.com/lmar76/my-python-repo/tests/test_bpack.py", line 23, in <module>
    record = BinaryRecord.frombytes(data)
  File "<string>", line 3, in frombytes
  File "/Users/luca mariani/Documents/github.com/lmar76/my-python-repo/venv/lib/python3.10/site-packages/bpack/codecs.py", line 236, in decode
    values = list(self._codec.unpack(data))
ValueError: Short data.
```

Executing it on Ubuntu, it raises a `bitstruct.Error`:

```
Python version: 3.10.8 (main, Oct 11 2022, 11:35:05) [GCC 11.2.0]
bpack version: 0.8.2
bitstruct version: 8.15.1

Record size: 16

Data has length: 2
{'field_1': 2, 'field_2': 26}

Data has length: 2
{'field_1': 2, 'field_2': 28}

Data has length: 1
Traceback (most recent call last):
  File "/home/ubuntu/Workspace/test_bpack.py", line 23, in <module>
    record = BinaryRecord.frombytes(data)
  File "<string>", line 3, in frombytes
  File "/home/ubuntu/Workspace/venv/lib/python3.10/site-packages/bpack/codecs.py", line 236, in decode
    values = list(self._codec.unpack(data))
  File "/home/ubuntu/Workspace/venv/lib/python3.10/site-packages/bitstruct/__init__.py", line 407, in unpack
    return self.unpack_from(data, allow_truncated=allow_truncated)
  File "/home/ubuntu/Workspace/venv/lib/python3.10/site-packages/bitstruct/__init__.py", line 428, in unpack_from
    return tuple([v[1] for v in self.unpack_from_any(
  File "/home/ubuntu/Workspace/venv/lib/python3.10/site-packages/bitstruct/__init__.py", line 428, in <listcomp>
    return tuple([v[1] for v in self.unpack_from_any(
  File "/home/ubuntu/Workspace/venv/lib/python3.10/site-packages/bitstruct/__init__.py", line 298, in unpack_from_any
    raise Error(
bitstruct.Error: unpack requires at least 16 bits to unpack (got 8)
```

Executing it on Windows, it raises a `bitstruct.Error`:

```
Python version: 3.10.8 | packaged by conda-forge | (main, Nov  4 2022, 13:42:51) [MSC v.1916 64 bit (AMD64)]
bpack version: 0.8.2
bitstruct version: 8.11.1

Record size: 16

Data has length: 2
{'field_1': 2, 'field_2': 26}

Data has length: 2
{'field_1': 2, 'field_2': 28}

Data has length: 1
Traceback (most recent call last):
  File "C:\Users\Administrator\test_bpack.py", line 23, in <module>
    record = BinaryRecord.frombytes(data)
  File "<string>", line 3, in frombytes
  File "C:\ProgramData\Anaconda3\envs\test_bpack\lib\site-packages\bpack\codecs.py", line 236, in decode
    values = list(self._codec.unpack(data))
  File "C:\ProgramData\Anaconda3\envs\test_bpack\lib\site-packages\bitstruct\__init__.py", line 400, in unpack
    return self.unpack_from(data)
  File "C:\ProgramData\Anaconda3\envs\test_bpack\lib\site-packages\bitstruct\__init__.py", line 421, in unpack_from
    return tuple([v[1] for v in self.unpack_from_any(data, offset)])
  File "C:\ProgramData\Anaconda3\envs\test_bpack\lib\site-packages\bitstruct\__init__.py", line 421, in <listcomp>
    return tuple([v[1] for v in self.unpack_from_any(data, offset)])
  File "C:\ProgramData\Anaconda3\envs\test_bpack\lib\site-packages\bitstruct\__init__.py", line 298, in unpack_from_any
    raise Error(
bitstruct.Error: unpack requires at least 16 bits to unpack (got 8)
```

**Comment by avalentino** [comment]
Dear @lmar76, thanks for reporting.
I assume that the issue is the different behaviour and the different exception type on different platforms.

After a short investigation it seems that the different behaviour is due to `bitstruct`.

On some platforms `bitstruct` can provide optional C extension to accelerate the computation.
Errors generated by the C extension can look different form the ones generated by the pure Python version of `bitstruct`.

To replicate the same behaviour on all platforms you could try to add
```python
del bitstruct.c
```
just after the the last import in your script.

By the way I do not recommend to do it for 2 reasons:

* it is a hack tat could have unknown side effects
* you will loose the benefits of the C extension acceleration.

In conclusion, I don't think that there is anything that I can do in `bpack` to improve the situation apart for providing a mechanism to disable the use of the `bitstruct` C extension.
If you think that you could need it please open a dedicated feature request issue (possibly referring to this one).

**Comment by avalentino** [comment]
Hi @lmar76, how should we proceed?
Do you need some change or can we close?

**Comment by avalentino** [comment]
The recommended solution is to catch both 'bitstruct.Error' and 'ValueError' (thanks to @lmar76).
